### PR TITLE
[MM-57991] Add log message about deleted rows from Data Retention job

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Server changes for https://github.com/mattermost/enterprise/pull/1670

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57991

#### Release Note
```release-note
Log information about deleted rows from Data Retention job
```

### CODE REVIEW

File: server/channels/store/sqlstore/retention_policy_store.go [hunk 1]
```diff
@@ -1152,9 +1152,16 @@ func getDeleteQueriesForMySQL(r RetentionPolicyBatchDeletionInfo, query string)
...
+	sqlResult, err := txn.Exec("DELETE FROM RetentionIdsForDeletion WHERE Id = ?", id)
+	if err != nil {
+		return 0, errors.Wrap(err, "Failed to delete from RetentionIdsForDeletion")
 	}
-	return nil
+
+	rowsAffected, err := sqlResult.RowsAffected()
+	if err != nil {
+		return 0, errors.Wrapf(err, "unable to retrieve rows affected")
+	}
```
Comment (cpoile) [inline server/channels/store/sqlstore/retention_policy_store.go:1164]: Just one question: Is it possible to fail here and return an error when the deletion succeeded?

Comment (cpoile) [reply server/channels/store/sqlstore/retention_policy_store.go:1164]: Maybe another: in Calls we had problems with `RowsAffected` having implementation-specific behaviour, is there a danger of that here?

File: server/channels/store/sqlstore/retention_policy_store.go [hunk 2]
```diff
@@ -1152,9 +1152,16 @@ func getDeleteQueriesForMySQL(r RetentionPolicyBatchDeletionInfo, query string)
...
-	if _, err := txn.Exec("DELETE FROM RetentionIdsForDeletion WHERE Id = ?", id); err != nil {
-		return errors.Wrap(err, "Failed to delete from RetentionIdsForDeletion")
+func deleteFromRetentionIdsTx(txn *sqlxTxWrapper, id string) (int64, error) {
+	sqlResult, err := txn.Exec("DELETE FROM RetentionIdsForDeletion WHERE Id = ?", id)
+	if err != nil {
+		return 0, errors.Wrap(err, "Failed to delete from RetentionIdsForDeletion")
 	}
-	return nil
+
+	rowsAffected, err := sqlResult.RowsAffected()
```
Comment (BenCookie95) [inline server/channels/store/sqlstore/retention_policy_store.go:1161]: `rowsAffected` will always be 1 here. You want to get the number of rows deleted in `DeleteOrphanedRowsByIds` in the reaction_store

**Review by BenCookie95 — CHANGES_REQUESTED** [review]

File: server/channels/store/sqlstore/retention_policy_store.go (hunk 2 above)
Comment (hanzei) [reply server/channels/store/sqlstore/retention_policy_store.go]: Good catch! Fixed via 4f81faa6f4fe33369bd11a646effcf588324abfe.

File: server/channels/store/sqlstore/retention_policy_store.go (hunk 1 above)
Comment (hanzei) [reply server/channels/store/sqlstore/retention_policy_store.go]: Good questions @cpoile. I'm not sure if I have the answers to these.


> Maybe another: in Calls we had problems with `RowsAffected` having implementation-specific behaviour, is there a danger of that here?

`RowsAffected` is used in the code base without differentiating between different driver implementations. Would you happen to have a handy link to the conversations about this in Calls so I can take a deeper look?



> Just one question: Is it possible to fail here and return an error when the deletion succeeded?

In all other places of the codebase where `RowsAffected` is used, the method returns with an error if RowsAffected` errors. I have never seen `RowsAffected` fail. So, I assume we are as safe here as we are with other parts of the code base.

File: server/channels/store/storetest/post_store.go [hunk 3]
```diff
@@ -4015,8 +4015,9 @@ func testPostStorePermanentDeleteBatch(t *testing.T, rctx request.CTX, ss store.
 	require.Equal(t, 1, len(rows))
 	require.Equal(t, 2, len(rows[0].Ids))
 	// Clean up retention ids table
-	err = ss.Reaction().DeleteOrphanedRowsByIds(rows[0])
+	deleted, err = ss.Reaction().DeleteOrphanedRowsByIds(rows[0])
 	require.NoError(t, err)
+	require.Equal(t, int64(0), deleted)
```
Comment (hanzei) [inline server/channels/store/storetest/post_store.go:4020]: Note to self: Why is this cleanup needed, if no reaction get deleted?

Comment (hanzei) [reply server/channels/store/storetest/post_store.go:4020]: This call is need to clean the post from RetentionIdsForDeletion by calling https://github.com/mattermost/mattermost/blob/a3ae02e266172fdf48e63890eca9c734291f85ca/server/channels/store/sqlstore/retention_policy_store.go#L1155-L1162

**Comment by hanzei** [comment]
@cpoile @BenCookie95 The PR is up to date with `master` and ready for review. PTAL.

**Comment by hanzei** [comment]
Unrelated CI issue

**Review by BenCookie95 — APPROVED** [review]
LGTM, let's ask QA to do a quick regression test before merging.

File: server/channels/store/sqlstore/retention_policy_store.go (hunk 1 above)
Comment (hanzei) [reply server/channels/store/sqlstore/retention_policy_store.go]: Thanks for sharing that additional piece of context @cpoile. It seems like we are good here, right?

**Review by cpoile — APPROVED** [review]
Thanks everyone!

**Review by DHaussermann — APPROVED** [review]
Tested and passe
- Confirmed that the job now writes `debug` level logs.
- Records show `Finished deleting rows from the Post table` as well as `ChannelMemberHistory`, `ThreadMembership` and `Thread table`
- The logging includes data about the job that performed the deletes
- Tested this a few times with different types data to delete.

No issues found 
LGTM!

**Pull request merged**